ports/atmel-samd/peripherals/samd21/events.c

```diff
@@ -183,31 +102,17 @@ bool event_interrupt_active(uint8_t channel) {
             EVSYS->INTFLAG.reg = EVSYS_INTFLAG_EVD(value) | EVSYS_INTFLAG_OVR(value);
         }
     }
-    #endif
-    #ifdef SAMD51
-    active = EVSYS->Channel[channel].CHINTFLAG.bit.EVD;
-    // Only clear if we know its active, otherwise there is the possibility it becomes after we
-    // check but before we clear.
-    if (active) {
-        EVSYS->Channel[channel].CHINTFLAG.reg = EVSYS_CHINTFLAG_EVD | EVSYS_CHINTFLAG_OVR;
-    }
-    #endif
     return active;
 }
 
 bool event_interrupt_overflow(uint8_t channel) {
     bool overflow = false;
-    #ifdef SAMD21
     if (channel > 7) {
```

Comment:
Off-by-1 error here for channels >= 8 (I had noticed this while debugging `PDMIn`. If, say, channel == 8, it will shift the channel number by 1 instead of by 0.
Change to:
```
    if (channel >= 8) {
        uint8_t value = 1 << (channel - 8);
```
Same issue at lines 156-157 and lines 171-172, but I can't leave comments there.